eta/core/c3d.py

```diff
@@ -0,0 +1,340 @@
+        conv5 = self.conv3d('conv5b', conv5, _weights['wc5b'], _biases['bc5b'])
+        conv5 = tf.nn.relu(conv5, 'relu5b')
+        pool5 = self.max_pool('pool5', conv5, k=2)
+
+        # Fully connected layer
+        pool5 = tf.transpose(pool5, perm=[0,1,4,2,3])
+        dense1 = tf.reshape(pool5, [1, _weights['wd1'].get_shape().as_list()[0]]) # Reshape conv3 output to fit dense layer input
+        dense1 = tf.matmul(dense1, _weights['wd1']) + _biases['bd1']
+
+        self.fc6 = tf.nn.relu(dense1, name='fc1') # Relu activation
+        dense1 = tf.nn.dropout(self.fc6, _dropout)
+
+        dense2 = tf.nn.relu(tf.matmul(dense1, _weights['wd2']) + _biases['bd2'], name='fc2') # Relu activation
+        dense2 = tf.nn.dropout(dense2, _dropout)
+
+        # Output: class prediction
+        self.out = tf.matmul(dense2, _weights['out']) + _biases['out']
+
+    def _load_model(self):
+        saver = tf.train.Saver()
+        init = tf.global_variables_initializer()
+        self.sess.run(init)
+        saver.restore(self.sess, self.config.model_path)
+
+
+class C3DFeaturizerConfig(C3DConfig):
+    ''' C3D Featurization configuration settings that works on images'''
+    pass
+
+
+class C3DFeaturizer(Featurizer):
+    def __init__(self, config=None):
+        super(C3DFeaturizer,self).__init__()
+        self.config = config or C3DFeaturizerConfig.default()
+        self.validate(self.config)
+        self.sample_method = self.config.sample_method
+        self.c3d  = None
+
+    def get_first_k_frames(self, inpath, k, out_size):
+        '''Sample first k frames in a video.
+
+        Args:
+            inpath: path to the input video
+            k: number of frames to extract for the input video
+            out_size: size of the output frames
+        '''
+        data = []
+        num_frames = etav.get_frame_count(inpath)
+        assert k < num_frames
+        with etav.VideoProcessor(inpath, "1-%d" % k, out_size=out_size) as p:
+            for img in p:
+                img = etai.resize(img, out_size, out_size)
+                data.append(img)
+        np_arr_data = np.array(data).astype(np.float32)
+        return np_arr_data
+
+    def uniformly_sample_k_frames(self, inpath, k, out_size):
+        '''Uniformly sample k frames, always including the first frame.
+
+        Args:
+            inpath: path to the input video
+            k: number of frames to extract for the input video
+            out_size: size of the output frames
+        '''
+        data = []
+        num_frames = etav.get_frame_count(inpath)
+        assert k < num_frames
+
+        stride = int(math.ceil(1.0 * num_frames / k))
+        rng = range(num_frames)
+        rng = list(rng[::stride])
+        if len(rng) < k:
+            rng.append(num_frames - 1)
+        elif len(rng) > k:
+            rng.pop()
+        assert len(rng) == k
+        with etav.VideoProcessor(inpath, out_size=out_size) as p:
+            for counter, img in enumerate(p):
```

Comment:
The proper way to only read a subset of frames is to pass the list of frames you want as the `frames` argument of `FFmpegVideoReader`.

Read the code you're calling before reimplementing something that already there!

Reply:
OK. I was using VideoProcessor before, don't see it can input a list. Yeah, FFmpegVideoReader makes life easier.